MCP broken (java.lang.NullPointerException: data) since bumping MCP client library in #1859

https://github.com/enola-dev/enola/actions/runs/18327398183/job/52195114808?pr=1859 for #1859,

fails running https://docs.enola.dev/use/mcp:

    ./enola -v mcp call-tool modelcontextprotocol/everything echo '{"message":"hi"}'

with this obscure error, but only on CI, not reproducible locally:

```
2025-10-07 22:12:07 SEVERE dev.enola.cli.common.LoggingMixin executionStrategy Hi! 👋 I'm https://enola.dev/ 29784cb-dirty. 👽 Resistance 👾 is futile. We are ONE. What's your goal, today?
2025-10-07 22:12:07 INFO dev.enola.cli.ExecMdCommand call Processing /home/runner/work/enola/enola/docs/use/ai/index.md...
2025-10-07 22:12:07 INFO dev.enola.cli.ExecMdCommand call Processing /home/runner/work/enola/enola/docs/use/mcp/index.md...
2025-10-07 22:12:07 INFO ch.vorburger.exec.ManagedProcess startPreparation Starting Program [/usr/bin/env, bash, -c, cd ../.././.. && ./enola -v mcp call-tool modelcontextprotocol/everything echo '{"message":"hi"}'] (in working directory /home/runner/work/enola/enola/docs/use/mcp)
2025-10-07 22:12:07 INFO ch.vorburger.exec.ManagedProcess waitForExitMaxMs Thread is now going to wait max. 7000ms for process to terminate itself: Program [/usr/bin/env, bash, -c, cd ../.././.. && ./enola -v mcp call-tool modelcontextprotocol/everything echo '{"message":"hi"}'] (in working directory /home/runner/work/enola/enola/docs/use/mcp)
2025-10-07 22:12:08 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 2025-10-07 22:12:08 SEVERE dev.enola.cli.common.LoggingMixin executionStrategy Hi! 👋 I'm https://enola.dev/ 29784cb-dirty. 👽 Resistance 👾 is futile. We are ONE. What's your goal, today?
2025-10-07 22:12:08 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: java.lang.NullPointerException: data
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at java.base/java.util.Objects.requireNonNull(Objects.java:259)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at reactor.core.publisher.FluxHandle$HandleSubscriber.next(FluxHandle.java:251)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at io.modelcontextprotocol.spec.McpClientSession.lambda$sendRequest$12(McpClientSession.java:280)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:113)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at reactor.core.publisher.SerializedSubscriber.onNext(SerializedSubscriber.java:99)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.onNext(FluxTimeout.java:181)
2025-10-07 22:12:09 SEVERE ch.vorburger.exec.SLF4jLogOutputStream processLine env: 	at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:176)
	at io.modelcontextprotocol.spec.McpClientSession.lambda$sendRequest$12(McpClientSession.java:280)
	at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:113)
	at reactor.core.publisher.SerializedSubscriber.onNext(SerializedSubscriber.java:99)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.onNext(FluxTimeout.java:181)
	at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:176)
	at io.modelcontextprotocol.spec.McpClientSession.handle(McpClientSession.java:157)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.onNext(FluxPeekFuseable.java:489)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2571)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.request(FluxPeekFuseable.java:437)
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:136)
	at reactor.core.publisher.FluxFlatMap$FlatMapInner.onSubscribe(FluxFlatMap.java:968)
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onSubscribe(FluxContextWrite.java:101)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.onSubscribe(FluxPeekFuseable.java:471)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4576)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:430)
	at reactor.core.publisher.SinkManyUnicast.drainRegular(SinkManyUnicast.java:284)
	at reactor.core.publisher.SinkManyUnicast.drain(SinkManyUnicast.java:365)
	at reactor.core.publisher.SinkManyUnicast.tryEmitNext(SinkManyUnicast.java:239)
	at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100)
	at io.modelcontextprotocol.client.transport.StdioClientTransport.lambda$startInboundProcessing$6(StdioClientTransport.java:254)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
		at reactor.core.publisher.Mono.block(Mono.java:1779)
		at io.modelcontextprotocol.client.McpSyncClient.ping(McpSyncClient.java:218)
		at dev.enola.ai.mcp.McpLoader.createSyncClient(McpLoader.java:200)
		at dev.enola.ai.mcp.McpLoader.lambda$opt$0(McpLoader.java:127)
		at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
		at dev.enola.ai.mcp.McpLoader.opt(McpLoader.java:127)
		at dev.enola.common.name.NamedTypedObjectProvider.get(NamedTypedObjectProvider.java:30)
		at dev.enola.ai.mcp.cli.CallToolCommand.call(CallToolCommand.java:76)
		at dev.enola.ai.mcp.cli.CallToolCommand.call(CallToolCommand.java:42)
		at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
		at picocli.CommandLine.access$1500(CommandLine.java:148)
		at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
		at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
		at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
		at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
		at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
		at dev.enola.cli.common.LoggingMixin.executionStrategy(LoggingMixin.java:77)
		at picocli.CommandLine.execute(CommandLine.java:2174)
		at dev.enola.cli.common.CLI.execute(CLI.java:93)
		at dev.enola.cli.EnolaApplication.main(EnolaApplication.java:64)
	at ch.vorburger.exec.ManagedProcess.waitForExitMaxMsWithoutLog(ManagedProcess.java:493)
	at ch.vorburger.exec.ManagedProcess.waitForExitMaxMs(ManagedProcess.java:470)
	at dev.enola.common.exec.vorburger.VorburgerExecRunner.exec(VorburgerExecRunner.java:60)
	at dev.enola.common.exec.vorburger.Runner.bash(Runner.java:52)
	at dev.enola.common.markdown.exec.ExecMD.exec(ExecMD.java:150)
	... 15 more
Caused by: java.util.concurrent.ExecutionException: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at ch.vorburger.exec.ManagedProcess.waitForExitMaxMsWithoutLog(ManagedProcess.java:478)
	... 19 more
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:355)
	at org.apache.commons.exec.DefaultExecutor.lambda$execute$0(DefaultExecutor.java:269)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Error: Process completed with exit code 1.
```

--- a/java/dev/enola/ai/mcp/McpLoader.java
+++ b/java/dev/enola/ai/mcp/McpLoader.java
@@ -31,7 +31,6 @@ import dev.enola.common.Version;
 import dev.enola.common.io.object.ObjectReader;
 import dev.enola.common.io.object.jackson.JacksonObjectReaderWriterChain;
 import dev.enola.common.io.resource.ReadableResource;
-import dev.enola.common.jackson.ObjectMappers;
 import dev.enola.common.name.NamedTypedObjectProvider;
 import dev.enola.common.secret.SecretManager;
 
@@ -41,7 +40,7 @@ import io.modelcontextprotocol.client.transport.HttpClientSseClientTransport;
 import io.modelcontextprotocol.client.transport.HttpClientStreamableHttpTransport;
 import io.modelcontextprotocol.client.transport.ServerParameters;
 import io.modelcontextprotocol.client.transport.StdioClientTransport;
-import io.modelcontextprotocol.json.jackson.JacksonMcpJsonMapper;
+import io.modelcontextprotocol.json.McpJsonMapper;
 import io.modelcontextprotocol.spec.McpClientTransport;
 import io.modelcontextprotocol.spec.McpSchema;
 
@@ -131,6 +130,11 @@ public class McpLoader implements NamedTypedObjectProvider<McpSyncClient> {
         }
     }
 
+    // TODO Make mcpJsonMapper() private instead of public when CallToolCommand doesn't need it
+    public McpJsonMapper mcpJsonMapper() {
+        return McpJsonMapper.getDefault();
+    }
+
     @VisibleForTesting
     McpServerConnectionsConfig.ServerConnection replaceSecretPlaceholders(
             McpServerConnectionsConfig.ServerConnection originalServerConnection)
@@ -150,9 +154,7 @@ public class McpLoader implements NamedTypedObjectProvider<McpSyncClient> {
         switch (connectionConfig.type) {
             case stdio -> {
                 var params = createStdIoServerParameters(connectionConfig);
-                var transport =
-                        new StdioClientTransport(
-                                params, new JacksonMcpJsonMapper(ObjectMappers.INSTANCE));
+                var transport = new StdioClientTransport(params, mcpJsonMapper());
                 transport.setStdErrorHandler(new McpServerStdErrLogConsumer(origin));
                 return transport;
             }

--- a/java/dev/enola/ai/mcp/cli/CallToolCommand.java
+++ b/java/dev/enola/ai/mcp/cli/CallToolCommand.java
@@ -25,10 +25,8 @@ import dev.enola.common.io.resource.ClasspathResource;
 import dev.enola.common.io.resource.FileResource;
 import dev.enola.common.io.resource.ResourceProvider;
 import dev.enola.common.io.resource.ResourceProviders;
-import dev.enola.common.jackson.ObjectMappers;
 import dev.enola.common.secret.auto.AutoSecretManager;
 
-import io.modelcontextprotocol.json.jackson.JacksonMcpJsonMapper;
 import io.modelcontextprotocol.spec.McpSchema.CallToolRequest;
 
 import org.jspecify.annotations.Nullable;
@@ -77,8 +75,7 @@ public class CallToolCommand implements Callable<Integer> {
         var callToolRequest =
                 CallToolRequest.builder()
                         .name(tool)
-                        .arguments(
-                                new JacksonMcpJsonMapper(ObjectMappers.INSTANCE), argumentsAsJson)
+                        .arguments(loader.mcpJsonMapper(), argumentsAsJson)
                         .build();
         var callToolResult = toolClient.callTool(callToolRequest);
         for (var content : callToolResult.content()) {

